video/out/opengl/context_drm_egl.c

```diff
@@ -671,10 +682,24 @@ static void drm_egl_uninit(struct ra_ctx *ctx)
 {
     struct priv *p = ctx->priv;
     struct drm_atomic_context *atomic_ctx = p->kms->atomic_context;
-
-    int ret = drmModeAtomicCommit(p->kms->fd, atomic_ctx->request, 0, NULL);
+    int flags = 0;
+    int ret = 0;
+
+     if (p->hdr_data.blob_id != 0) {
+        p->hdr_data.new_csp = (struct mp_colorspace){0};
+        ret = drm_create_hdr_metadata(atomic_ctx, &p->hdr_data);
+        if (ret) {
+            MP_WARN(ctx->vo, "Failed to reset HDR state on display (%d)\n", ret);
+        } else {
+            drm_object_set_property(atomic_ctx->request, atomic_ctx->connector,
+                                    "HDR_OUTPUT_METADATA", p->hdr_data.blob_id);
+            flags |= DRM_MODE_ATOMIC_ALLOW_MODESET;
```

Comment:
Personally, I would always create a SDR blob + other (as needed) during the init. Then during teardown, I would reuse the SDR one to "clean-up" or reset the state and ultimately destroy all blobs.

Note: I'm not 100% sure if we need to reset the HDR state on teardown - it's been a while and my memory is fuzzy.

Reply:
It definitely needs to be reset, the way I do it now it will check if the color information changes and update SDR/HDR depending on that.

Reply:
My (seemingly failed) emphasis was to create the SDR blob at the start (alongside the HDR one) but only use it at teardown/exit. Cleaner code (init path creates 2 blobs, uninit patch destroys 2 blobs) and we don't risk being "stuck" in HDR, if the SDR blob creation fails.

Reply:
Ah, I see what you mean. That is certainly an option for the SDR one, but again for the HDR one I don't think that will work, as we discussed above. Admittedly I'm also a little fuzzy on where, exactly, things need to happen.